packages/calcite-components/src/utils/form.tsx

```diff
@@ -263,20 +287,27 @@ function invalidHandler(event: Event) {
   // dispatch a "calciteInvalid" so users can set custom validation messages
   formComponent.dispatchEvent(new CustomEvent("calciteInvalid", { bubbles: true, composed: true }));
 
+  if ("validity" in formComponent) {
```

Comment:
Should `validity` be set before dispatching the event?

Reply:
Nice catch, I must have switched them around fixing merge conflicts